gpii/node_modules/lifecycleManager/src/LifecycleManager.js

```diff
@@ -1309,4 +1319,94 @@ var gpii = fluid.registerNamespace("gpii");
         return fluid.promise.sequence(tasks);
     };
 
+    /**
+     * Function to add the current name of the preferences set to the full
+     * payload which is output from the matchmaking process.
+     *
+     * @param {Component} that - An instance of LifecycleManager component
+     * @param {Object} fullPayload - The full structure output from the matchmaking process. NOTE: This object will be modified
+     * @param {String} [forcePrefsSetName] - [optional] A preferences set name to be selected.
+     * @return {Object} The modified fullPayload, now including activePrefsSetName, activeConfiguration
+     */
+    gpii.lifecycleManager.updateActivePrefsSetName = function (that, fullPayload, forcePrefsSetName) {
+        var newActivePrefsSetName = "gpii-default";
+        if (forcePrefsSetName === undefined) {
+            if (fullPayload.forcePrefsSetName) { // Don't use stored if name has been forced
+                newActivePrefsSetName = fullPayload.activePrefsSetName;
+            } else {
+                var activeSession = that.getSession();
+                if (activeSession.model.activePrefsSetName) {
+                    newActivePrefsSetName = activeSession.model.activePrefsSetName;
+                }
+            }
+        } else {
+            newActivePrefsSetName = forcePrefsSetName;
+            fullPayload.forcePrefsSetName = true;
+        }
+        fullPayload.activePrefsSetName = newActivePrefsSetName;
+        return fullPayload;
+    };
+
+    /**
+     * On changes to the preferences set:
+     * 1) parse the set against the currently logged in user's conditions
+     * 2) if the newly calculated active prefsSet is different from the currently applied prefsSet,
+     *    the LifecycleManager's update functionality will be called and the applied prefsSet and
+     *    configuration will be stored in the active session of the LifecycleManager.
+     *
+     * @param {Component} that - The lifecycleManager component from which the current
+     *    session will be retrieved
+     * @param {String} [forcePrefsSetName] - [optional] A prefsSet name to be unconditionally selected
+     * @return {Promise} If the preferences set change causes an actual change in the system (i.e. change of
+     *    activePrefsSet and changes to the applied configuration) a promise of an update to the system
+     *    configuration is returned. If the call to prefsSetChanged does not result in a different
+     *    active preferences set or no user is logged in, undefined is returned.
+     **/
+    gpii.lifecycleManager.prefsSetNameChanged = function (that, forcePrefsSetName) {
```

Comment:
As now we only need to support the context change request from PSPChannel, the logic of this function can be cleaned up and simplified:

1. start with verifications:
1.1 if no one is logged in, or, a system reserved key is logged in, do nothing
1.2 if the new prefs set is not provided (`!newPrefsSetName`) or it's same as the active prefs set (`activeSession.model.activePrefsSetName === newPrefsSetName`), do nothing

2. update the payload: set prefs set name; delete appliedSolutions; calculate the new lifecycle instruction. Also, It's no longer necessary to keep `gpii.lifecycleManager.updateActivePrefsSetName()` function. It can be replaced by one single assignment - `fullPayload.activePrefsSetName = newPrefsSetName`

3. apply updated payload: run `lifecycleManager.update(payload)` and update activeSession.model in its promise callback.

Please also clean up comments that no longer make sense.

Reply:
> 1.2 if the new prefs set is not provided (!newPrefsSetName) or it's same as the active prefs set (activeSession.model.activePrefsSetName === newPrefsSetName), do nothing

What is `newPrefsSetName`?  Do you mean the local`newActivePrefsSetName`?  Or, do you mean to replace `forcePrefsSetName` with `newPrefsSetName`?

Also, there's a difference between this new logic and the old in that the old logic could result in a new prefs set name of `gpii-default`.  In the old logic, this happened when:
- no new name is passed in (`forcePrefsSetName` is undefined)
- `activeSession.model.activePrefsSetName` is undefined or null

See the beginning of [`gpii.lifecycleManager.updateActivePrefsSetName()`](https://github.com/klown/universal/blob/GPII-3117/gpii/node_modules/lifecycleManager/src/LifecycleManager.js#L1332), where the new name is assigned a default of `gpii-default`. Is a default new name of `gpii-default` still a possibility?

Reply:
>What is newPrefsSetName? Do you mean the localnewActivePrefsSetName? Or, do you mean to replace forcePrefsSetName with newPrefsSetName?
The `newPrefsSetName` I meant was the new prefsSetName that comes in from PSPChannel requests.

In terms of the variable name, "newPrefsSetName" seems better than "forcePrefsSetName".

>there's a difference between this new logic and the old in that the old logic could result in a new prefs set name of gpii-default.
Yes, the default pref set name is "gpii-default". I believe the old logic that auto-switches to "gpii-default" was created to handle the case when the environment change doesn't fall into any pref set. In the new logic, it would be cleaner to have the new function expect a valid new prefs set name and simply switch to it.

Reply:
>  Also, It's no longer necessary to keep gpii.lifecycleManager.updateActivePrefsSetName() function.

I just removed it and tests are failing.  It's used (at least) by the `gpii.lifecycleManager.userLogonHandling.stateChangeHandler` on a `onMatchDone` event.  See https://github.com/klown/universal/blob/GPII-3117/gpii/node_modules/lifecycleManager/src/UserLogonStateChange.js#L206.  I'm looking into how to handle this, but if you have any insights, that would be helpful.

Reply:
Good catch, @klown. ya, seems we need to keep this function hopefully with a cleaner logic.